lib/checks/navigation/identical-links-same-purpose.js

```diff
@@ -0,0 +1,151 @@
+/**
+ * Note: `identical-links-same-purpose-after` fn, helps reconcile the results
+ */
+const { text } = axe.commons;
+
+/**
+ * Note:
+ * Firefox does not respect map > area as visible,
+ * even when used by a img[usemap], hence the usage of `includeHidden` flag,
+ * in the accessible name computation
+ */
+const accText = text.accessibleTextVirtual(virtualNode, {
+	includeHidden: true
+});
+const name = curateText(accText, {
+	emoji: true,
+	nonBmp: true,
+	punctuations: true,
+	sanitize: true,
+	lowercase: true
+});
+if (!name) {
+	return undefined;
+}
+
+/**
+ * Set `data` and `relatedNodes` for use in `after` fn
+ */
+const afterData = {
+	name,
+	parsedResource: getParsedResource(node, 'href')
+};
+this.data(afterData);
+this.relatedNodes([node]);
+
+return true;
+
+/**
+ * Curate a given string
+ * @method curateText
+ * @param {String} str given string to curate
+ * @param {Object} options options to curate
+ * @property {Boolean} options.emoji remove emoji characters?
+ * @property {Boolean} options.nonBmp remove nonBmp characters?
+ * @property {Boolean} options.punctuations remove punctuation characters?
+ * @property {Boolean} options.sanitize santize given string?
+ * @property {Boolean} options.lowercase convert given string to lowercase?
+ * @returns {String}
+ */
+function curateText(str, options) {
+	const {
+		emoji = false,
+		nonBmp = false,
+		punctuations = false,
+		sanitize = false,
+		lowercase = false
+	} = options;
+
+	let curatedText = text.removeUnicode(str, { emoji, nonBmp, punctuations });
+
+	if (sanitize) {
+		curatedText = text.sanitize(curatedText);
+	}
+
+	if (lowercase) {
+		curatedText = curatedText.toLowerCase();
+	}
+
+	return curatedText;
+}
+
+/**
+ * Construct a resource object for a given node fromt he attribute provided
+ * @method getParsedResource
+ * @param {HTMLElement} currentNode node
+ * @param {String} attribute of the node to be used for resolving resource
+ * @returns {Object}
+ */
+function getParsedResource(currentNode, attribute) {
+	const value = currentNode[attribute];
+	if (!value) {
+		return undefined;
+	}
+
+	const nodeName = currentNode.nodeName.toUpperCase();
+	let parser = currentNode;
+
+	if (!['A', 'AREA'].includes(nodeName)) {
+		parser = document.createElement('a');
+		parser.href = value;
+	}
+
+	const [pathname, filename] = getPathnameAndFilename(parser.pathname);
+	return {
+		protocol: parser.protocol,
+		hostname: parser.hostname,
+		port: parser.port,
+		pathname: pathname.replace(/^\/|\/$/g, ''), // revmove lead/trial(ing) slashes
+		search: parser.search,
+		hash: getHash(parser.hash),
```

Comment:
Can we rename these to hashRoute and getHashRoute? This isn't just the hash after all.